### DISCUSSION TITLE
setOptions for lean({virtuals: true})

### DISCUSSION THREAD

**Discussion by skcc75** [Automattic/mongoose#10355] [Q&A] [answered]
We have the same code which applies `lean` to all mongoose calls.

```
mongoose.Query.prototype.setOptions = function(options, overwrite) {
  __setOptions.apply(this, arguments);
  if (this.options.lean == null) this.mongooseOptions({lean: true});
  return this;
};
```

What's the way to change it to enable virtuals for `lean`?

**Comment by vkarpov15** [comment, marked as answer]
`if (this.options.lean == null) this.mongooseOptions({lean: {virtuals:true}});`

Also you would need to use the [mongoose-lean-virtuals plugin](https://www.npmjs.com/package/mongoose-lean-virtuals)
